rs/replicated_state/src/canister_state.rs

```diff
@@ -109,7 +109,8 @@ impl SchedulerState {
 }
 
 /// The full state of a single canister.
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug)]
+#[cfg_attr(debug_assertions, derive(PartialEq))]
```

Comment:
Most tests are run specified under `cfg(test)`, and the fact that `debug_assertions` is enabled is just a cargo/bazel config. So `#[cfg_attr(any(test, debug_assertions), derive(PartialEq))]` would be more accurate.

Reply:
We compile dependencies for tests without `cfg(test)`, so it's irrelevant. I chose `debug_assertions` because it's the config for long and slow checks, what `PageMap` `eq` is